test/CP/core/search/lns.jl

```diff
@@ -121,9 +121,329 @@
+        model.limit.searchingTime = 2
+
+        x = SeaPearl.IntVar(1, 2, "x", trailer)
+        y = SeaPearl.IntVar(1, 2, "y", trailer)
+        SeaPearl.addVariable!(model, x)
+        SeaPearl.addVariable!(model, y)
+        SeaPearl.addConstraint!(model, SeaPearl.Equal(x, y, trailer))
+        SeaPearl.addObjective!(model, x)
+
+        search = SeaPearl.LNSearch()
+        status = SeaPearl.expandLns!(search, model, SeaPearl.MinDomainVariableSelection(), SeaPearl.BasicHeuristic()) 
+        @test model.limit.searchingTime < 2
+
+        # Ensure that time limits passed as argument are correctly managed (nothing)
+        trailer = SeaPearl.Trailer()
+        model = SeaPearl.CPModel(trailer)
+
+        x = SeaPearl.IntVar(1, 2, "x", trailer)
+        y = SeaPearl.IntVar(1, 2, "y", trailer)
+        SeaPearl.addVariable!(model, x)
+        SeaPearl.addVariable!(model, y)
+        SeaPearl.addConstraint!(model, SeaPearl.Equal(x, y, trailer))
+        SeaPearl.addObjective!(model, x)
+
+        search = SeaPearl.LNSearch(limitIterNoImprovement=1, limitValuesToRemove = 2)
+        status = SeaPearl.expandLns!(search, model, SeaPearl.MinDomainVariableSelection(), SeaPearl.BasicHeuristic()) 
+        @test model.limit.searchingTime === nothing
+
+        # Ensure that time limits passed as argument are correctly managed (localSearchTimeLimit and globalTimeLimit)
+        trailer = SeaPearl.Trailer()
+        model = SeaPearl.CPModel(trailer)
+        model.limit.searchingTime = 10
+
+        x = SeaPearl.IntVar(1, 100, "x", trailer)
+        y = SeaPearl.IntVar(1, 100, "y", trailer)
+        z = SeaPearl.IntVar(1, 100, "z", trailer)
+        SeaPearl.addVariable!(model, x)
+        SeaPearl.addVariable!(model, y)
+        SeaPearl.addVariable!(model, z)
+        SeaPearl.addConstraint!(model, SeaPearl.Equal(x, y, trailer))
+        SeaPearl.addConstraint!(model, SeaPearl.Equal(y, z, trailer))
+        SeaPearl.addObjective!(model, z)
+        
+        search = SeaPearl.LNSearch(limitIterNoImprovement=20000, limitValuesToRemove = 3, repairLimits=Dict("searchingTime" => 10))
+        status = SeaPearl.expandLns!(search, model, SeaPearl.MinDomainVariableSelection(), SeaPearl.BasicHeuristic()) 
+        @test model.limit.searchingTime < 10
+
+        # Ensure that `limitValuesToRemove` works properly
+        trailer = SeaPearl.Trailer()
+        model = SeaPearl.CPModel(trailer)
+        model.limit.searchingTime = 3
+
+        x = SeaPearl.IntVar(1, 2, "x", trailer)
+        y = SeaPearl.IntVar(1, 2, "y", trailer)
+        SeaPearl.addVariable!(model, x)
+        SeaPearl.addVariable!(model, y)
+        SeaPearl.addConstraint!(model, SeaPearl.Equal(x, y, trailer))
+        SeaPearl.addObjective!(model, x)
+
+        search = SeaPearl.LNSearch(limitValuesToRemove = 1)
+        status = SeaPearl.expandLns!(search, model, SeaPearl.MinDomainVariableSelection(), SeaPearl.BasicHeuristic()) 
+        @test status === :NonOptimal
```

Comment:
Can you recall why the result is not Optimal ?

Reply:
Because the first solution found (during initialization) will be {x => 2, y => 2} (because `BasicHeuristic` choose the biggest value in the domain). This solution is non optimal as x is the objective variable and his domain is {1, 2}. As we set `limitValuesToRemove` to 1, the destroy and repair loop will never reach any solution (because of the constraint `Equal(x, y, trailer)`). In that way, the solution returned will be {x => 2, y => 2} (≠ to optimal solution {x => 1, y => 1}).

Reply:
I will add some comments in test file

Reply:
Ok thanks, that's perfectly clear!